crates/sui-tool/src/lib.rs

```diff
@@ -174,83 +158,107 @@ impl std::fmt::Display for OwnerOutput {
     }
 }
 
-pub struct GroupedObjectOutput(pub ObjectData);
+#[allow(clippy::type_complexity)]
+pub struct GroupedObjectOutput {
+    pub grouped_results: BTreeMap<
+        Option<(
+            Option<SequenceNumber>,
+            ObjectDigest,
+            TransactionDigest,
+            Owner,
+            Option<TransactionDigest>,
+        )>,
+        Vec<AuthorityName>,
+    >,
+    pub voting_power: Vec<(
+        Option<(
+            Option<SequenceNumber>,
+            ObjectDigest,
+            TransactionDigest,
+            Owner,
+            Option<TransactionDigest>,
+        )>,
+        u64,
+    )>,
+    pub available_voting_power: u64,
+    pub fully_locked: bool,
+}
+
+impl GroupedObjectOutput {
+    pub fn new(
+        object_data: ObjectData,
+        committee: Arc<BTreeMap<AuthorityPublicKeyBytes, u64>>,
+    ) -> Self {
+        let mut grouped_results = BTreeMap::new();
+        let mut voting_power = BTreeMap::new();
+        let mut available_voting_power = 0;
+        for (name, _, (version, resp, _elapsed)) in &object_data.responses {
+            let stake = committee.get(name).unwrap();
+            let key = match resp {
+                Ok(r) => {
+                    let obj_digest = r.object.compute_object_reference().2;
+                    let parent_tx_digest = r.object.previous_transaction;
+                    let owner = r.object.owner;
+                    let lock = r.lock_for_debugging.as_ref().map(|lock| *lock.digest());
+                    if lock.is_none() {
+                        available_voting_power += stake;
+                    }
+                    Some((*version, obj_digest, parent_tx_digest, owner, lock))
+                }
+                Err(_) => None,
```

Comment:
Should available_voting_power increment in this case as well?

Reply:
it depends on the exact scenario, leaving it as-is for now for simplicity